lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -186,6 +188,17 @@ def add_3ds(post, options)
         end
       end
 
+      def add_processing_channel(post, options)
+        post[:processing_channel_id] = options[:processing_channel_id] if options[:processing_channel_id]
+      end
+
+      def add_marketplace_data(post, options)
+        if options[:marketplace]
+          post[:marketplace] = {}
+          post[:marketplace][:sub_entity_id] = options[:marketplace][:sub_entity_id] if options[:marketplace][:sub_entity_id]
```

Comment:
Let's check for nil with `options[:marketplace]` before attempting to dig into it

Reply:
@jessiagee that is done on [line 196](https://github.com/activemerchant/active_merchant/pull/4236/files#diff-ac1e1ae60a6cc04111367e448e705b6e74104fd908e64d1ffe6c7237b5deb65dR196).

Reply:
All good @jessiagee, thanks for the review.